src/internal/connector/onedrive/folder_cache.go

```diff
@@ -0,0 +1,28 @@
+package onedrive
+
+import (
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// TODO: refactor to comply with graph/cache_container
```

Comment:
Do we need this TODO? It seems like `cache_container` does a lot more

Reply:
maybe, maybe not.  It's not a high priority concern.

Reply:
Of note here, from Georgi's point, lookups by folder name are a little vulnerable.  It'd be nice to run this as a two step enumerate-then-check cache of folders, much like exchange does with CacheContainer.